ah/kubernetes_node_pools.go

```diff
@@ -0,0 +1,146 @@
+	Count             int               `json:"count,omitempty"`
+	MinCount          int               `json:"min_count,omitempty"`
+	MaxCount          int               `json:"max_count,omitempty"`
+	AutoScale         bool              `json:"autoscale,omitempty"`
+}
+
+// UpdateKubernetesNodePoolRequest represents a request to update a node pool
+type UpdateKubernetesNodePoolRequest struct {
+	Labels    map[string]string `json:"labels,omitempty"`
+	Name      string            `json:"name,omitempty"`
+	Count     int               `json:"count,omitempty"`
+	AutoScale bool              `json:"autoscale,omitempty"`
+	MinCount  int               `json:"min_count,omitempty"`
+	MaxCount  int               `json:"max_count,omitempty"`
+}
+
+// GetNodePool returns node pool
+func (kc *ClustersService) GetNodePool(ctx context.Context, clusterId, nodePoolId string) (*KubernetesNodePool, error) {
+	path := fmt.Sprintf("api/v2/kubernetes/clusters/%s/node_pools/%s", clusterId, nodePoolId)
+	req, err := kc.client.newRequest(http.MethodGet, path, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var nodePoolRoot KubernetesNodePoolRoot
+
+	if _, err = kc.client.Do(ctx, req, &nodePoolRoot); err != nil {
+		return nil, err
+	}
+
+	return nodePoolRoot.KubernetesNodePool, nil
+}
+
+// ListNodePools returns list of node pools
+func (kc *ClustersService) ListNodePools(ctx context.Context, options *ListOptions, clusterId string) ([]KubernetesNodePool, error) {
+	path := fmt.Sprintf("api/v2/kubernetes/clusters/%s/node_pools", clusterId)
+
+	var NodePoolsRoot KubernetesNodePoolsRoot
+
+	if err := kc.client.list(ctx, path, options, &NodePoolsRoot); err != nil {
+		return nil, err
+	}
+
+	return NodePoolsRoot.KubernetesNodePools, nil
+}
+
+// CreateNodePool creates node pool
+func (kc *ClustersService) CreateNodePool(ctx context.Context, clusterId string, request *CreateKubernetesNodePoolRequest) (*KubernetesNodePool, error) {
+	path := fmt.Sprintf("api/v2/kubernetes/clusters/%s/node_pools", clusterId)
+	req, err := kc.client.newRequest(http.MethodPost, path, request)
+	if err != nil {
+		return nil, err
+	}
+
+	var nodePoolRoot KubernetesNodePoolRoot
+	if _, err := kc.client.Do(ctx, req, &nodePoolRoot); err != nil {
+		return nil, err
+	}
+
+	return nodePoolRoot.KubernetesNodePool, nil
+}
+
+// UpdateNodePool updates node pool
+func (kc *ClustersService) UpdateNodePool(ctx context.Context, clusterId, nodePoolId string, request *UpdateKubernetesNodePoolRequest) error {
+	path := fmt.Sprintf("api/v2/kubernetes/clusters/%s/node_pools/%s", clusterId, nodePoolId)
+	req, err := kc.client.newRequest(http.MethodPut, path, request)
+	if err != nil {
+		return err
+	}
+
+	if _, err := kc.client.Do(ctx, req, nil); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// DeleteNodePool deletes node pool
+func (kc *ClustersService) DeleteNodePool(ctx context.Context, clusterId string, nodePoolId string, replace bool) error {
```

Comment:
minor: why `kc` and not `cs` at least? it looks a bit confusing to me

Reply:
Yes, but in [kubernetes_clusters](https://github.com/advancedhosting/advancedhosting-api-go/blob/master/ah/kubernetes_clusters.go#L86) for ClustersService use kc

Reply:
good point and might be my bad, I'll create a separate task for renaming